### ISSUE TITLE
MeshDepthMaterial + MeshNormalMaterial should take into account displacement, normals, bump parameters

### ISSUE THREAD

**Issue by bhouston** [mrdoob/three.js#8551]
##### Description of the problem 

MeshDepthMaterial and MeshNormalMaterial only uses the basic triangle information and not any of the microsurface modifications: displacement, normal and bump.  This leads to the resulting packed normals being less accurate and representative.

For MeshDepthMaterial it is critical that it utilize the displacement settings as this moves the triangles around.

For MeshNormalMaterial it should, for best results, utilize properly the original material's bumpMap/Scale, normalMap/Scale, as well as displacementMap/Scale/Bias.

Basically I think this can be done partially by having override materials utilize parameters from the original materials if there is a match, or we could special case it.

Together this will lead to more accurate post effects, such as SSAO, SAO, DOF, etc.

##### Three.js version

- [x] Dev
- [x] r75
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] Linux
- [ ] Android
- [ ] IOS

**Comment by bhouston** [comment]
ACtually I think that this bug is a lot of work to fix in WebGL 1.0 but it will be easy to address, and very efficient, in WebGL 2.0.  Closing now.
